arm-containerregistry/2016-06-27-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,642 @@
+          "readOnly": true
+        },
+        "reason": {
+          "description": "The reason that the container registry name could not be used.",
+          "type": "string",
+          "readOnly": true
+        },
+        "message": {
+          "description": "The error message that explains the reason value in more detail.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "Registry": {
+      "description": "The container registry.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/RegistryProperties",
+          "description": "The container registry properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "RegistryProperties": {
+      "description": "The container registry properties.",
+      "properties": {
+        "loginServer": {
+          "description": "The URL to log into the container registry.",
+          "type": "string",
+          "readOnly": true
+        },
+        "creationDate": {
+          "format": "date-time",
+          "description": "The creation date of the container registry in ISO8601 format.",
+          "type": "string",
+          "readOnly": true
+        },
+        "adminUserEnabled": {
+          "description": "The boolean value that indicates whether admin user is enabled.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccountBase",
+          "description": "The storage account base properties.",
+          "readOnly": true
+        }
+      }
+    },
+    "StorageAccountBase": {
+      "description": "The storage account base properties.",
+      "properties": {
+        "name": {
+          "description": "Storage account name.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RegistryCreateParameters": {
+      "description": "The parameters for creating a container registry.",
+      "required": [
+        "location",
+        "properties"
+      ],
+      "properties": {
+        "location": {
+          "description": "Resource location.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/RegistryPropertiesCreateParameters",
+          "description": "The parameters for creating container registry properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "RegistryPropertiesCreateParameters": {
+      "description": "The parameters for creating container registry properties.",
+      "required": [
+        "tenantId",
+        "storageAccount"
+      ],
+      "properties": {
+        "tenantId": {

```

Comment:
We are planning to revamp the storage management spec. It has not been modeled accurately. So having that as a reference check is not ideal.

Yes, I was thinking about StorageAccount. As per the current model accessKey, endPointUrl and location are _write-only_ (that is they can be created and updated) but cannot be read. _Currently swagger does not provide a mechanism to specify write-only_. 
Actually, only **accessKey** contains sensitive information. What is the harm in sending the **endPointUrl** and **location** back in the response? 
Ideally StorageAccountBase  should have `{ name, location, endPointUrl }` as **required** properties.
StorageAccountProperties { accessKey (required) } should inherit from StorageAccountBase.

**RegistryProperties.storageAccount** should be of  type **StorageAccountProperties**. So when the service sends the response it will not have accessKeys, which I think should be fine as we do not validate on deserialization.

So having Registry as input to PUT and output of PUT, you will make customers life easy and open up the CreateOrUpdate() scenario on PUT. 

The input model for PATCH as an update is good.